Source/CombatExtended/CombatExtended/SiegeUtility.cs

```diff
@@ -0,0 +1,85 @@
+using System.Collections.Generic;
+using System.Linq;
+using RimWorld;
+using Verse;
+using Verse.AI.Group;
+
+namespace CombatExtended
+{
+    [StaticConstructorOnStartup]
+    public class SiegeUtility
+    {
+        /// <summary>
+        /// The minimum required construction skill to be able to build any potential siege artillery in the game.
+        /// </summary>
+        public static readonly int MinRequiredConstructionSkill;
+
+        static SiegeUtility()
+        {
+            MinRequiredConstructionSkill = DefDatabase<ThingDef>.AllDefsListForReading
```

Comment:
Out of curiosity, how does vanilla handle making sure they spawn at least one pawn with high enough Construction to build the mortar? Do they have a dynamic check, or do they just make sure at least one has 5 construction (which is what mortars require)?